🐛 Bug: refresh doesn't work when `.Provider.TokenValidation = IdToken`

### Reproduction steps

_No response_

### Expected vs actual behavior

Error:
```
[ERROR] [traefik-oidc-auth] Failed to validate renewed session: token has invalid claims: token is expired
```

Which leads to a logout every time an access token expires

Probably we need to add `IdToken` to a session here
https://github.com/sevensolutions/traefik-oidc-auth/blob/8a46e4771d6a95a1c90cf75795bfdc49e6438b6b/session.go#L111-L114

### Environment and Version Numbers

- Traefik v3.3.4
- Plugin v0.10.0
- IDP: Keycloak 2.26

All running in Kubernetes, middeware configured like this:

<details>
<summary>Middleware configuration</summary>

```yaml
apiVersion: traefik.io/v1alpha1
kind: Middleware
metadata:
  name: my-middleware
spec:
  plugin:
    oidc-auth:
      Authorization:
        AssertClaims:
        - AnyOf:
          - my-group
          Name: groups
      CookieNamePrefix: my_prefix
      Provider:
        ClientId: [REDACTED]
        ClientSecret: [REDACTED]
        Url: [REDACTED]
        UsePkce: true
        TokenValidation: IdToken // Works with Introspection
      Scopes:
      - openid
      - email
      - profile
      - groups
      - offline_access
      Secret: [REDACTED]
      SessionCookie:
        Domain: [REDACTED]
        MaxAge: 86400
        SameSite: lax
```

</details>

### Log Output

_No response_

### Doing your homework

- [x] I have searched for similar bug reports and not found a similar one.

--- a/.github/ISSUE_TEMPLATE/bug.yml
+++ b/.github/ISSUE_TEMPLATE/bug.yml
@@ -34,7 +34,7 @@ body:
         * Any special circumstances, like self-signed TLS certificates, or similar?
       placeholder: |-
         * Traefik v3.3.2
-        * plugin version v0.10.0
+        * plugin version v0.10.1
         * IDP: Dex v2.41.0
         All running on Docker.
         Dex is fronted by Traefik on a dedicated subdomain, using a LetsEncrypt cert.

--- a/session.go
+++ b/session.go
@@ -107,10 +107,20 @@ func validateSessionTicket(toa *TraefikOidcAuth, encryptedTicket string) (*sessi
 				return nil, nil, nil, err
 			}
 
-			toa.logger.Log(logging.LevelInfo, "Successfully renewed session")
-
 			session.AccessToken = newTokens.AccessToken
 			session.RefreshToken = newTokens.RefreshToken
+
+			// We had some problems with some providers which didn't return a new IdToken when renewing the tokens.
+			// Thats why i'am logging this case specifically here.
+			if newTokens.IdToken != "" {
+				session.IdToken = newTokens.IdToken
+			} else {
+				if toa.Config.Provider.TokenValidation == "IdToken" {
+					toa.logger.Log(logging.LevelWarn, "The auth provider didn't return a new IdToken. Still keeping the old one.")
+				} else {
+					toa.logger.Log(logging.LevelDebug, "The auth provider didn't return a new IdToken. Still keeping the old one.")
+				}
+			}
 
 			success, claims, err = toa.validateToken(session)
 
@@ -118,6 +128,8 @@ func validateSessionTicket(toa *TraefikOidcAuth, encryptedTicket string) (*sessi
 				toa.logger.Log(logging.LevelError, "Failed to validate renewed session: %v", err)
 				return nil, nil, session, err
 			}
+
+			toa.logger.Log(logging.LevelInfo, "Successfully renewed session")
 
 			return session, claims, session, err
 		} else {

--- a/traefik-config.yml
+++ b/traefik-config.yml
@@ -22,7 +22,7 @@ experimental:
   #plugins:
   #  traefik-oidc-auth:
   #    moduleName: "github.com/sevensolutions/traefik-oidc-auth"
-  #    version: v0.10.0
+  #    version: v0.10.1
   
 providers:
   file:

--- a/website/docs/getting-started/index.md
+++ b/website/docs/getting-started/index.md
@@ -16,7 +16,7 @@ experimental:
   plugins:
     traefik-oidc-auth:
       moduleName: "github.com/sevensolutions/traefik-oidc-auth"
-      version: "v0.10.0"
+      version: "v0.10.1"
 ```
 
 ## Configure Middleware

